ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -494,4 +510,37 @@ private void displayWelcomeMessage() {
 
     writer.flush();
   }
+
+  static void maybeCreateProcessingLogStream(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig,
+      final KsqlEngine ksqlEngine,
+      final CommandQueue commandQueue) {
+    if (!config.getString(ProcessingLogConfig.STREAM_AUTO_CREATE).equals(
+        ProcessingLogConfig.AUTO_CREATE_ON)) {
+      return;
+    }
+    final String name = config.getString(ProcessingLogConfig.STREAM_NAME);
+    final String topicName = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    final String statementText = ProcessingLogServerUtils.processingLogStreamCreateStatement(
+        name,
+        topicName);
+    if (!commandQueue.isEmpty()) {
+      return;
+    }
+    final List<PreparedStatement<?>> statement;
+    try {
+      statement = ksqlEngine.parseStatements(statementText);
+      ksqlEngine.tryExecute(statement, ksqlConfig, Collections.emptyMap());
+    } catch (final KsqlException e) {
+      log.warn("Failed to create processing log stream", e);
+      return;
+    }
+    commandQueue.enqueueCommand(
```

Comment:
The creation of this log stream will count toward the active persistent query limit. Not sure whether this is preferable to making an exception for it or not, but wanted to throw that out there.

Reply:
It's just a ddl statement, so no query gets started.

Reply:
Ah yes, silly me. Guess the past few days were more tiring than I thought, heh.